astropy/nddata/nddata_base.py

```diff
@@ -60,33 +57,13 @@ def wcs(self):
     @abstractproperty
     def meta(self):
         """
-        Metadata, if any, must be dict-like.
+        Meta information, if any.
         """
         return None
 
-    # uncertainty and its setter are implemented as concrete to enforce the
-    # logic in the uncertainty setter. For a long discussion of the problems
-    # with trying to implement them as abstract (particularly the setter but
-    # not the getter), see http://bugs.python.org/issue11610
-    #
-    # In python >= 3.3 it would be easy to decorate one of these (setter or
-    # getter) as abstract but not the other.
-    @property
+    @abstractproperty
```

Comment:
@embray -- you mentioned elsewhere that there is a way to enforce the requirement that `uncertainty` have an `uncertainty_type` that is a string without providing a concrete implementation in `NDDataBase`...could you point us to some references?